File: src/main/java/com/polygloat/dtos/response/ApiKeyDTO/ApiKeyDTO.java
package com.polygloat.dtos.response.ApiKeyDTO;

import com.polygloat.constants.ApiScope;
import com.polygloat.model.ApiKey;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Set;
import java.util.stream.Collectors;

@Data
@Builder
@Schema
@NoArgsConstructor
@AllArgsConstructor
public class ApiKeyDTO {
    private Long id;

    @Schema(name = "Resulting user's api key")
    private String key;

    private String userName;

    private Long repositoryId;

    private String repositoryName;

    private Set<String> scopes;

    public static ApiKeyDTO fromEntity(ApiKey apiKey) {
        return ApiKeyDTO.builder()
                .key(apiKey.getKey())
                .id(apiKey.getId())
                .userName(apiKey.getUserAccount().getName())
                .repositoryId(apiKey.getRepository().getId())
                .repositoryName(apiKey.getRepository().getName())
                .scopes(apiKey.getScopes().stream().map(ApiScope::getValue).collect(Collectors.toSet()))
                .build();
    }
}

File: src/main/kotlin/io/tolgee/dtos/response/ApiKeyDTO/ApiKeyDTO.kt
package io.tolgee.dtos.response.ApiKeyDTO

import io.swagger.v3.oas.annotations.media.Schema
import io.tolgee.model.ApiKey
import io.tolgee.model.enums.ApiScope
import java.util.stream.Collectors

class ApiKeyDTO(
  var id: Long = 0,
  @field:Schema(description = "Resulting user's api key")
  var key: String = "",
  var userName: String? = null,
  var projectId: Long = 0,
  var projectName: String = "",
  var scopes: Set<String> = setOf()
) {

  companion object {
    fun fromEntity(apiKey: ApiKey): ApiKeyDTO {
      return ApiKeyDTO(
        key = apiKey.key,
        id = apiKey.id,
        userName = apiKey.userAccount.name,
        projectId = apiKey.project.id,
        projectName = apiKey.project.name,
        scopes = apiKey.scopesEnum.stream().map(ApiScope::value).collect(Collectors.toSet())
      )
    }
  }
}
